rust-codegen/codegen/src/parser.rs

```diff
@@ -0,0 +1,450 @@
+use std::str;
```

Comment:
Is there a reason why you went with your own parser implementation instead of using the official one? Having two separate parsers that have to be kept in sync can result in subtle parsing bugs between the two implementations.